internal/controller/datadogagent/profile.go

```diff
@@ -137,11 +133,21 @@ func setProfileDDAIAffinity(ddai *v1alpha1.DatadogAgentInternal, profile *v1alph
 	override.Affinity = common.MergeAffinities(override.Affinity, agentprofile.AffinityOverride(profile))
 }
 
-func setProfileDDAIMeta(ddai *v1alpha1.DatadogAgentInternal, profile *v1alpha1.DatadogAgentProfile, scheme *runtime.Scheme) error {
+func setProfileDDAIMeta(ddai *v1alpha1.DatadogAgentInternal, profile *v1alpha1.DatadogAgentProfile) error {
 	// Name
 	ddai.Name = getProfileDDAIName(ddai.Name, profile.Name, profile.Namespace)
 	// Managed fields
 	ddai.ManagedFields = []metav1.ManagedFieldsEntry{}
+	// Include the profile label in the DDAI metadata only for non-default profiles.
+	// This is used to determine whether or not the EDS should be created (only for default profile).
+	// This could possibly be used for GC of the profile DDAIs too.
+	if !agentprofile.IsDefaultProfile(profile.Namespace, profile.Name) {
+		// This should not happen, as we add the DDA label upon creation of the DDAI.
+		if ddai.Labels == nil {
+			ddai.Labels = make(map[string]string)
+		}
```

Comment:
Just in case, but we should always have label (DDA name) from the DDA controller